### ISSUE TITLE
When adding a link, both the "float bar" and the "Insert link" dialog open

### ISSUE THREAD

**Issue by klonos** [backdrop-contrib/ckeditor5#17]
1. Add the link button in the active toolbar for a CKE5-enabled text format.
2. Start creating content in a type with a field that allows using that format.
3. Type some text.
4. Select some of the text, and click the "Link" button in the CKE toolbar

Result:
<img width="1187" alt="image" src="https://github.com/quicksketch/ckeditor5/assets/2423362/f93d1802-82dc-4a9e-a0ae-93d32aac6bfa">

The expectation I guess is to either:
- have only the Backdrop dialog pop up, and suppress the native floating dialog provided by CKE
- or only show the native floating dialog provided by CKE, and then have the "Insert link" dialog pop up when the "Advanced" button is clicked on the native dialog.

**Comment by quicksketch** [comment]
Yeah, this is an interesting question. I personally don't like trying to use the floating dialog (CKEditor calls this the "balloon" interface) for large amounts of configuration. You can see what https://www.drupal.org/project/editor_advanced_link looks like with nearly all the same options we have:

![image](https://github.com/quicksketch/ckeditor5/assets/100206/c9d24449-20da-4f03-b99d-16b1bd1e7707)

But we also have the File Upload, which couldn't convert so easily to using a balloon. Overall I feel like using Backdrop-provided modals gives us more flexibility and extensibility (at least without getting neck-deep in CKEditor custom code).

All that aside, CKEditor's built-in link plugin really assumes you're going to be using the Balloon UI. And if we didn't use CKEditor's built-in link plugin, then we have to do all the work of figuring out where links should be inserted, how they should wrap elements, checking for nested images, etc. It's a lot.

I tried just simply hiding the balloon as soon as it pops up. That seems to mostly work but the editor seems to loose cursor position sometimes. If you want to try out both hiding the balloon and not automatically opening the dialog, check for these lines in in the `backdrop-link.js` file:

```
        // Immediately click the newly added button.
        this.button.fire('execute');

        // @todo: Find a way to close the balloon but maintain focus.
        // linkUI._hideUI();
```

You can comment out the first line to see what it's like manually opening the Backdrop modal; you'll need to click "Advanced" in the balloon to open it. Or un-comment out the second line to automatically hide the balloon and check if you see any side-effects.

**Comment by quicksketch** [comment]
A small update here: after working on Image handling in #1, I found that perhaps instead of using the out-of-box link button, we could replace it with our own button. Previously the image button was using functionality similar to CKEditor core, and now it opens a dialog directly. Perhaps we can use a similar approach to make the toolbar button open a dialog all the time, while selecting an existing link would use the balloon interface with an option to open the dialog.

**Comment by quicksketch** [comment]
I filed a PR at https://github.com/backdrop-contrib/ckeditor5/pull/77. This approach does end up creating our own UI buttons, but we mostly still use the logic and functionality of the built-in link plugin. We still need to modify the balloon toolbar manually, which is (still) pretty weird. In this version, rather than making a new balloon button, I took over the "Edit link" button, which now opens the Backdrop dialog:

![image](https://github.com/backdrop-contrib/ckeditor5/assets/100206/4a4e8931-27cc-4e0f-bd1c-fd14df95fb99)

**Comment by quicksketch** [comment]
Note if testing #77 (and if this gets merged): If you have CKEditor 5 on a text format, you'll need to edit the format to re-add the Link button to the toolbar, since the underlying button ID changed from `link` to `backdropLink`.

**Comment by indigoxela** [comment]
Oh yes, much better!

I tested with regular text links, with and without attributes, linking and unlinking, switching to source view, saving, opening a saved node's form – all that works properly. :+1: 

Keeping the balloon button has the advantage, that opening the link form doesn't require to scroll up to the toolbar to edit a link, even if the text is longer. :wink:

**Comment by quicksketch** [comment]
Well even though it's a lot of extra code, I feel like adding our own buttons is the "right" approach. We may change this further as we go but this set of changes is an improvement over the current behavior. I merged #77 into 1.x-1.x. Thanks again @indigoxela for reviewing and testing!
